src/sentry/integrations/aws_lambda/integration.py

```diff
@@ -0,0 +1,168 @@
+    IntegrationMetadata,
+    FeatureDescription,
+)
+from sentry.models import Project, ProjectKey
+from sentry.pipeline import PipelineView
+from sentry.shared_integrations.exceptions import IntegrationError
+from sentry.web.helpers import render_to_response
+from sentry.utils.compat import filter
+
+from .client import gen_aws_lambda_client
+from .utils import parse_arn
+
+logger = logging.getLogger("sentry.integrations.aws_lambda")
+
+DESCRIPTION = """
+The AWS Lambda integration will automatically instrument your Lambda functions without any code changes. All you need to do is run a CloudFormation stack that we provide to get started.
+"""
+
+
+FEATURES = [
+    FeatureDescription(
+        """
+        Instrument your serverless code automatically.
+        """,
+        IntegrationFeatures.SERVERLESS,
+    ),
+]
+
+metadata = IntegrationMetadata(
+    description=DESCRIPTION.strip(),
+    features=FEATURES,
+    author="The Sentry Team",
+    noun=_("Installation"),
+    issue_url="https://github.com/getsentry/sentry/issues/new",
+    source_url="https://github.com/getsentry/sentry/tree/master/src/sentry/integrations/aws_lambda",
+    aspects={},
+)
+
+SUPPORTED_RUNTIMES = ["nodejs12.x", "nodejs10.x"]
+
+
+class AwsLambdaIntegration(IntegrationInstallation):
+    pass
+
+
+class AwsLambdaIntegrationProvider(IntegrationProvider):
+    key = "aws_lambda"
+    name = "AWS Lambda"
+    requires_feature_flag = True
+    metadata = metadata
+    integration_cls = AwsLambdaIntegration
+    features = frozenset([IntegrationFeatures.SERVERLESS])
+
+    def get_pipeline_views(self):
+        return [AwsLambdaCloudFormationPipelineView(), AwsLambdaSetupLayerPipelineView()]
+
+    def build_integration(self, state):
+        # TODO: unhardcode
+        integration_name = "Serverless Hack Bootstrap"
+
+        arn = state["arn"]
+        parsed_arn = parse_arn(arn)
+        account_id = parsed_arn["account"]
+        region = parsed_arn["region"]
+
+        external_id = u"{}-{}".format(account_id, region)
+
+        integration = {
+            "name": integration_name,
+            "external_id": external_id,
+            "metadata": {"arn": state["arn"], "aws_external_id": state["aws_external_id"]},
+        }
+        return integration
+
+
+class AwsLambdaCloudFormationPipelineView(PipelineView):
+    def dispatch(self, request, pipeline):
+        # TODO: Find way so a user/org only gets one external id
+        if request.method == "POST":
+            arn = request.POST["arn"]
+            aws_external_id = request.POST["aws_external_id"]
+            pipeline.bind_state("arn", arn)
+            pipeline.bind_state("aws_external_id", aws_external_id)
+            return pipeline.next_step()
+
+        template_url = options.get("aws-lambda.cloudformation-url")
+        aws_external_id = uuid.uuid4()
+
+        cloudformation_url = (
+            "https://console.aws.amazon.com/cloudformation/home#/stacks/create/review?"
```

Comment:
@scefali now that we are on python3 we can use f-strings. %-formatting is not recommended for new code.

Reply:
@NisanthanNanthakumar so using `.format` you mean?